### PR TITLE
feat(widget): Adds Key/Value widget

### LINKED ISSUE #50: Create widget for key-value parameters
For advanced properties in lot of our components and cluster definitions we will need key-value parameters. a key in a technical name of a properties, this key have default value.

The key field is a list but the use should be able to enter manually a custom properties that is not in the list

image

### PR DESCRIPTION
**Pull request by glorieux**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/Talend/react-talend-forms/blob/master/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

The forms library didn't include a key/value widget.

**What is the new behavior?**

The forms library contains a key/value widget.

**Other information**:

* Adds Key/Value Widget
* Adds keyValue to the list of available uiSchema Widget
* Adds Key/Value stories

Fixes #50

### ScreenShots
Don't represent final rendering.

#### Key/Value Array
screen shot 2017-01-18 at 10 51 26

#### Key/Value
screen shot 2017-01-18 at 10 51 40

#### Key(enum)/Value Array
screen shot 2017-01-18 at 10 51 58

### CODE REVIEW

File: src/widgets/KeyValueWidget.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+			value={toValue(data)}
+			onChange={(change) => onChange(`${toKey(data)}=${change}`)}
+		/>
+	);
+}
+
+function KeyValueWidget(props) {
+	return (
+		<div className="key-value">
+			{ renderKey(props.schema, props.value || props.formData, props.onChange) }
```
Comment (frassinier) [inline src/widgets/KeyValueWidget.js]: These three args could be variables I guess?

File: src/widgets/KeyValueWidget.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		<Text
+			value={toValue(data)}
+			onChange={(change) => onChange(`${toKey(data)}=${change}`)}
+		/>
+	);
+}
+
+function KeyValueWidget(props) {
+	return (
+		<div className="key-value">
```
Comment (frassinier) [inline src/widgets/KeyValueWidget.js]: You could use Definition List `<dl>` and add `=` as a CSS pseudo-element?

File: stories/json/keyValueEnumArray.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    }
+  },
+  "uiSchema": {
+    "keyValue": {
+      "items": {
+        "ui:widget": "keyValue"
+      }
+    }
+  },
+  "properties": {}
```
Comment (frassinier) [inline stories/json/keyValueEnumArray.json]: Can you add at least one example with properties?

**Comment by jmfrancois** [comment]
Please put a screenshot of the widget as it is in the PR description

File: src/widgets/KeyValueWidget.js (hunk 1 above)
Comment (talend-glorieux) [inline src/widgets/KeyValueWidget.js]: I'm not a big fan of assigning variables when they are just used twice. Especially strings.

**Pull request merged**